libs/model/src/comment/DeleteComment.command.ts

```diff
@@ -0,0 +1,49 @@
+import { InvalidActor, type Command } from '@hicommonwealth/core';
+import * as schemas from '@hicommonwealth/schemas';
+import { models } from '../database';
+import { isAuthorized, type AuthContext } from '../middleware';
+import { mustBeAuthorized, mustExist } from '../middleware/guards';
+
+export function DeleteComment(): Command<
+  typeof schemas.DeleteComment,
+  AuthContext
+> {
+  return {
+    ...schemas.DeleteComment,
+    auth: [isAuthorized({})],
+    body: async ({ actor, payload, auth }) => {
+      const { address } = mustBeAuthorized(actor, auth);
+      const { comment_id, message_id } = payload;
+
+      const comment = await models.Comment.findOne({
+        where: message_id
+          ? { discord_meta: { message_id } }
+          : { id: comment_id },
+        include: [
+          {
+            model: models.Thread,
+            attributes: ['community_id'],
+            required: true,
+          },
+        ],
+        logging: true,
+      });
+      mustExist('Comment', comment);
+
+      if (comment.address_id !== address!.id && address.role === 'member')
```

Comment:
Shouldn't this be handled in `isAuthorized`? If this is handled in `isAuthorized` then we can also remove line 15 `mustBeAuthorized` from above.

Reply:
That's the ideal way to handle it, with the new actor flag, etc... but it breaks the discord bot integration.
So, I'm keeping the existing pattern of loading inside the body when discord bot shares the command. 

Once we release the API, we can think about how to optimize this pattern ... maybe by finding the entity_id in the bot layer before the command gets invoked, and we can clean all at once.